HIP/hip-859.md

```diff
@@ -0,0 +1,87 @@
+This value directly represents the gas consumed by the EVM, without taking into account the gas refund policy. This makes it possible to calculate the overall gas consumption for the transaction, during the ContractActions sidecar record importing.  
+
+If we sum up all of the `gasUsed` values of each ContractActions sidecar for a given transaction on top of the initial intrinsic gas, which is the initial gas, charged prior to the transaction execution, then we would get accurate value for the `gasConsumed` field.
+
+## User stories
+
+1. As a user, I want to see how much gas was consumed by the EVM during the execution of my transaction, when I retrieve a transaction info from the mirror-node REST APIs, so that I know if I can optimise my transaction cost
+2. As a user, I want to see and compare how much gas I was charged for and how much gas was consumed by the EVM, when I inspect my transaction in hashscan
+
+## Specification
+
+In order to calculate the `gasUsed` field, the mirror node should have `CONTRACT_ACTION` and `CONTRACT_BYTECODE` sidecars enabled. Then, we would use an enhancement logic in the `mirror-node-importer`,
+which will iterate over all `ContractAction` sidecar records for the transaction we want to import and calculate the `gasConsumed` field for.
+
+The new logic would have a mechanism to first calculate the intrinsic gas cost for the transaction being imported. We can inspect if the transaction has a `ContractByteCode` sidecar
+related to it. If yes, we have a contract creation transaction and we would calculate the intrinsic gas cost based on the length of the init bytecode. If we don't have such a sidecar,
+then the transaction represents a contract call and we would use the default intrinsic gas cost for contract calls, which is 21 000.
+
+On top of the intrinsic gas cost, we would iterate over all `ContractAction`
+sidecars and sum up the `gasUsed` field for each of them. The result would be the `gasConsumed` field for the transaction.
+
+The `gas_used` field in `contract_result` table would still have the same semantics and represent the gas that was charged to the user.
+
+## Backwards Compatibility
+
+This HIP we will extend the REST APIs responses related to transaction info with an additional field `gas_consumed`. This won't be a breaking change, but the users should have in mind the introduction of the new field.
+
+## Security Implications
+
+None
+
+## How to Teach This
+
+Respective documentation will be added.
+
+## Reference Implementation
+
+Please follow [this issue](https://github.com/hashgraph/hedera-mirror-node/issues/7543)
+
+## Rejected Ideas
+
+While discussing the implementation of this HIP, we considered the following alternative:
+
+Adding a new `gasConsumed` field to the Record Stream V6 format, which would be a breaking change in the protobuf format and would require a new version of the Record Stream to be introduced.
```

Comment:
I think there was another 1 or 2 ideas right?
One was calculating this all on the REST API side

Reply:
Added them as well